maplibre/src/render/error.rs

```diff
@@ -10,6 +10,11 @@ pub enum RenderError {
     Graph(#[from] RenderGraphError),
     #[error("error while requesting device")]
     RequestDevice(#[from] wgpu::RequestDeviceError),
+    #[cfg(feature = "headless")]
+    #[error("error while writing surface to png")]
+    WritePng(#[from] png::EncodingError),
+    #[error("error in file io")]
+    IoError(#[from] std::io::Error),
```

Comment:
Instead of adding these errors to the RenderError, a new error type should be created: HeadlessError/HeadlessRender error.

That also eliminates the need for the `cfg()`

Reply:
Why this is good idea is described here: https://mmapped.blog/posts/12-rust-error-handling.html#errors-problem-vs-solution

Reply:
I moved it separately. May you should have a look at the naming. I tried to make something more specific that `IoError`